[BUILD-1677]: Implement Network Policies for operand components

anchi205 opened the pull request:
Implement NetworkPolicies for operand components in the openshift-builds namespace to prevent lateral movement and data exfiltration. Added default-deny-all with explicit allow rules for K8s API egress, webhook ingress from openshift-kube-apiserver, and metrics ingress from openshift-monitoring. Includes an e2e test verifying both YAML deployment and actual network connectivity.

sayan-biswas commented on config/networkpolicies/10-egress-policies.yaml:
```diff
@@ -0,0 +1,15 @@
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: allow-k8s-api-egress
```
I believe, this policy should be a part of the bundle as this should be deployed by OLM not operator.

sayan-biswas commented on config/networkpolicies/90-default-deny-all.yaml:
```diff
@@ -0,0 +1,10 @@
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: default-deny-all
```
Same here, this policy should be a part of the bundle as this should be deployed by OLM not operator.

sayan-biswas commented on config/networkpolicies/20-ingress-policies.yaml:
```diff
@@ -0,0 +1,62 @@
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: shipwright-webhook-ingress
```
The webhook is also deployed by OLM and shouldn't be considered operand. Hence IMO, this should be also be part of the bundle.

sayan-biswas commented on config/networkpolicies/20-ingress-policies.yaml:
```diff
@@ -0,0 +1,62 @@
+    - protocol: TCP
+      port: 6000  # provisioner-m metrics port
+    - protocol: TCP
+      port: 9898  # healthz port
```
Metric port is different in shipwright controller in think.

sayan-biswas reviewed (changes requested):
There are few changes required as per this docs..

https://docs.google.com/document/d/1CDoGSRd-h8VT4PMrK_83Ro0YzYPjORbkxtfTJU1sN6Q/edit?tab=t.0

https://docs.google.com/document/d/1CDoGSRd-h8VT4PMrK_83Ro0YzYPjORbkxtfTJU1sN6Q/edit?tab=t.wpn5gzkaq9mg

anchi205 committed:
[BUILD-1677] feat(networkpolicy): Implement NetworkPolicies for operand components

Add ingress policies to openshift-builds: a default-deny for all pods,
webhook ingress from openshift-kube-apiserver (port 8443) for CSI and Shipwright
webhooks, and metrics ingress from monitoring-labeled namespaces for CSI (ports
6000, 9898) and Shipwright (port 8383). Wires a NetworkPolicy reconciler into the
operator controller and bundles the manifests into the image. Includes 17 e2e tests
covering spec validation and OVN enforcement (allow/deny probes on every policy).

Signed-off-by: Anchita Borah <anborah@redhat.com>
